internal/handlers/pg/pgdb/query.go

```diff
@@ -268,46 +269,81 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 			}
 		}
 
-		// Handle _id with a simpler query, as it can't be an array
-		if k == "_id" {
-			switch v := v.(type) {
-			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
-				// type not supported for pushdown
-				// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-				// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
-			case float64, string, types.ObjectID, int32, int64:
-				// Select if value under the key is equal to provided value.
-				sql := `(_jsonb->%[1]s)::jsonb = %[2]s`
-
-				// placeholder p.Next() returns SQL argument references such as $1, $2 to prevent SQL injections.
-				// placeholder $1 is used for field key,
-				// placeholder $2 is used for field value v.
-				filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next()))
-				args = append(args, k, string(must.NotFail(pjson.MarshalSingleValue(v))))
-
-			default:
-				panic(fmt.Sprintf("Unexpected type of value: %v", v))
-			}
+		// check if values are equal or if the left value contains the right one
+		eqOperator := "@>"
 
-			continue
+		if k == "_id" {
+			// check if values are equal
+			eqOperator = "="
 		}
 
 		switch v := v.(type) {
-		case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
+		case *types.Document:
+			iter := v.Iterator()
+			defer iter.Close()
+
+			for {
+				docKey, docVal, err := iter.Next()
+				if err != nil {
+					if errors.Is(err, iterator.ErrIteratorDone) {
+						break
+					}
+
+					return "", nil, lazyerrors.Error(err)
+				}
+
+				switch docKey {
+				case "$eq":
+					switch v := docVal.(type) {
+					case *types.Document, *types.Array, types.Binary, bool,
+						time.Time, types.NullType, types.Regex, types.Timestamp:
+						// type not supported for pushdown
+					case float64, string, types.ObjectID, int32, int64:
+						// Select if value under the key is equal to provided value.
+						sql := `(_jsonb->%[1]s)::jsonb ` + eqOperator + ` %[2]s`
```

Comment:
```suggestion
						sql := `(_jsonb->` + p.Next() + `)::jsonb ` + eqOperator + ` ` + p.Next()
```

Let's not mix string concatenation and sprintf.

And btw, why we need `::jsonb`?

Reply:
> And btw, why we need ::jsonb?

We don't... We use them from beginning, but now I see that there's no point of conversion into the jsonb type, when you select from jsonb column...

Reply:
> Let's not mix string concatenation and sprintf.

How do we want to do that in `$ne`?

```go
p1, p2 := p.Next(), p.Next()

sql := `NOT ( ` +
	// does document contain the key,
	// it is necessary, as NOT won't work correctly if the key does not exist.
	`_jsonb ? ` + p1 + ` AND ` +
	// does the value under the key is equal to the filter
	`_jsonb->` + p1 + ` ` + eqOperator + ` ` + p2 + ` AND ` +
	// does the value type is equal to the filter's one
	`_jsonb->'$s'->'p'->` + p1 + `->'t' = '"` + pjson.GetTypeOfValue(v) +
	`"')`
```

If we don't want to mix concatenation with sprintf, I would rather go the opposite way and include eqOperator to `Sprintf` values. What do you think?

Reply:
> Let's not mix string concatenation and sprintf.

If we can't use just concatenation, let use just sprintf